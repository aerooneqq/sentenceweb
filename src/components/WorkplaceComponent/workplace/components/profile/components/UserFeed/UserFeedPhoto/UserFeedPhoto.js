import React, {Component} from "react";

import "./UserFeedPhotoStyles.css";

//Services
import UserService from "../../../../../../../../services/UserServices/UserService";
import ResponseService from "../../../../../../../../services/ResponseService/ReponseService";
import ImageService from "../../../../../../../../services/ImageService/ImageService";

export default class UserFeedPhoto extends Component { 
    constructor(props) { 
        super(props);

        this.userService = new UserService(localStorage.getItem("token"));
        this.responseService = new ResponseService();

        this.state = {
            isLoading: false,
            userPhoto: null
        }

        this.getUserPhoto = this.getUserPhoto.bind(this);
    }

    getUserPhoto() { 
        return `data:image/png;base64, ${this.props.photo}`;
    }

    render() { 
        return ( 
            this.state.isLoading === true ? null :
                <img src = {this.getUserPhoto()} className = "singleUserFeedPhoto" alt = "avatar"/>
        )
    }
}