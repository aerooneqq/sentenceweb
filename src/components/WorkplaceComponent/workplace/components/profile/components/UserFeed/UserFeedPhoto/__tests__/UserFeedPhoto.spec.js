import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UserFeedPhoto from "../UserFeedPhoto";

describe("UserFeedPhoto", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<UserFeedPhoto />))).toMatchSnapshot();
    });

    it("should have proper initial state", () => { 
        const properState = { 
            isLoading: false,
            userPhoto: null
        } 

        const userFeedPhoto = shallow(<UserFeedPhoto />);

        for (let key in properState) { 
            expect(properState[key]).toBe(userFeedPhoto.state(key));
        }
    });

    it("should return proper user photo string from getUserPhoto method", () => { 
        const props = { 
            photo: "test string"
        }
        const userFeedPhoto = shallow(<UserFeedPhoto {...props}/>);

        expect(userFeedPhoto.instance().getUserPhoto()).toBe(`data:image/png;base64, ${props.photo}`);

    });
});