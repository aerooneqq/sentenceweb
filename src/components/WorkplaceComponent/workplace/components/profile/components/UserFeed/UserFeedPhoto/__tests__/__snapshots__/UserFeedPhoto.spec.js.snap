// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFeedPhoto should render correctly 1`] = `
<img
  alt="avatar"
  className="singleUserFeedPhoto"
  src="data:image/png;base64, undefined"
/>
`;
